stdlib/option.gr

```diff
@@ -1,78 +1,196 @@
+ * @param option: The option to unwrap
+ * @returns The unwrapped value if the Option is the `Some` variant or the default value otherwise
+ * 
+ * @since v0.2.0
+ */
+export let unwrapWithDefault = (default, option) => {
   match (option) {
     Some(x) => x,
     None => default
   }
 }
 
-let map = (fn, option) => {
+/**
+ * If the Option is `Some(value)`, applies the given function to the `value` and wraps the new value in a `Some` variant.
+ *
+ * @param fn: The function to call on the value of a `Some` variant
+ * @param option: The option to map
+ * @returns A new `Some` variant produced by the mapping function if the variant was `Some` or the unmodified `None` otherwise
+ * 
+ * @since v0.2.0
+ */
+export let map = (fn, option) => {
   match (option) {
     Some(x) => Some(fn(x)),
     None => None
   }
 }
 
-let mapWithDefault = (fn, default, option) => {
+/**
+ * If the Option is `Some(value)`, applies the given function to the `value` to produce a new value, otherwise uses the default value.
+ * Useful for unwrapping an Option while providing a fallback for any `None` variants.
+ *
+ * @param fn: The function to call on the value of a `Some` variant
+ * @param default: A fallback value for a `None` variant
+ * @param option: The option to map
+ * @returns The value produced by the mapping function if the Option is of the `Some` variant or the default value otherwise
+ * 
+ * @since v0.2.0
+ */
+export let mapWithDefault = (fn, default, option) => {
   match (option) {
     Some(x) => fn(x),
     None => default
   }
 }
 
-let mapWithDefaultFn = (fn, defaultFn, option) => {
+/**
+ * If the Option is `Some(value)`, applies the `fn` function to the `value` to produce a new value.
+ * If the Option is `None`, calls the `defaultFn` function to produce a new value.
+ * Useful for unwrapping an Option into a value, whether it is `Some` or `None`.
+ *
+ * @param fn: The function to call on the value of a `Some` variant
+ * @param defaultFn: The default function
+ * @param option: The option to map
+ * @returns The value produced by one of the mapping functions
+ * 
+ * @since v0.2.0
+ */
+export let mapWithDefaultFn = (fn, defaultFn, option) => {
   match (option) {
     Some(x) => fn(x),
     None => defaultFn()
   }
 }
 
-let flatMap = (fn, option) => {
+/**
+ * If the Option is `Some(value)`, applies the given function to the `value` to produce a new Option.
+ *
+ * @param fn: The function to call on the value of a `Some` variant
+ * @param option: The option to map
+ * @returns A new Option produced by the mapping function if the variant was `Some` or the unmodified `None` otherwise
+ * 
+ * @since v0.2.0
+ */
+export let flatMap = (fn, option) => {
   match (option) {
     Some(x) => fn(x),
     None => None
   }
 }
 
-let filter = (pred, option) => {
+/**
+ * Converts `Some(value)` variants to `None` variants where the predicate function returns `false`.
+ * Useful for applying some value validation to your Options.
```

Comment:
I'm not sure this makes a ton of sense or is the right way to think about `filter`.
```suggestion
```